File: user-app/lib/models/auth_response.dart
class AuthResponse {
  final String accessToken;
  final String tokenType;
  final int userId;
  final String email;
  final String fullName;
  final String role;

  AuthResponse({
    required this.accessToken,
    required this.tokenType,
    required this.userId,
    required this.email,
    required this.fullName,
    required this.role,
  });

  factory AuthResponse.fromJson(Map<String, dynamic> json) {
    return AuthResponse(
      accessToken: json['accessToken'],
      tokenType: json['tokenType'] ?? 'Bearer',
      userId: json['userId'],
      email: json['email'],
      fullName: json['fullName'],
      role: json['role'],
    );
  }
}

File: backend/src/main/java/com/servicebooking/dto/AuthResponse.java
package com.servicebooking.dto;

import com.servicebooking.model.UserRole;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class AuthResponse {
    private String accessToken;
    private String tokenType = "Bearer";
    private Long userId;
    private String email;
    private String fullName;
    private UserRole role;

    public AuthResponse(String accessToken, Long userId, String email, String fullName, UserRole role) {
        this.accessToken = accessToken;
        this.userId = userId;
        this.email = email;
        this.fullName = fullName;
        this.role = role;
    }
}
